File: Upcoder/Other/Ước số chung lớn nhất/LTCB-92.cs
using System;

namespace UCLN
{
    public class Program
    {
        static long UCLN(long n, long m)
        {
            if (m == 0)
                return n;
            return UCLN(m, n % m);
        }

        static void Main(string[] args)
        {
            string line;
            while ((line = Console.ReadLine()) != null)
            {
                string[] input = line.Split(' ');
                long n = long.Parse(input[0]);
                long m = long.Parse(input[1]);
                Console.WriteLine(UCLN(n, m));
            }
        }
    }
}

File: Upcoder/Other/Ước số chung lớn nhất/LTCB-92.py
def UCLN(n, m):
    if m == 0:
        return n
    return UCLN(m, n % m)

def main():
    while True:
        try:
            line = input()
            input_data = line.split(' ')
            n = int(input_data[0])
            m = int(input_data[1])
            print(UCLN(n, m))
        except EOFError:
            break
if __name__ == "__main__":
    main()
